### PR TITLE
Сleaned and updated the code 

### PR DESCRIPTION
**Pull request by IvashchenkoIvanIgorevich**
Сleaned and updated the code

### CODE REVIEW

File: CRM.Library.Services/HomeService/BookService.cs [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+﻿using DAL.Interfaces;
 using DAL.Models;
 using System.Collections.Generic;
 
 namespace CRMLibrary.Services.HomeService
 {
     public class BookService : IBookService
     {
-        public BookService(IGenericRepository<Book> repositoryBook)
+        public BookService(IBookRepository bookRepository)
```
Comment (tstrizhyk) [inline CRM.Library.Services/HomeService/BookService.cs]: why did you change IGenericRepository<Book> to IBookRepository?

File: CRM.Library.Services/HomeService/BookService.cs [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
         }
 
-        private IGenericRepository<Book> _repository;
+        public void Edit(Book editBook)
+        {
+            _repository.Update(editBook);
+            _repository.Complete();
+        }
+
+        public Book GetById(int? id)
```
Comment (tstrizhyk) [inline CRM.Library.Services/HomeService/BookService.cs]: I thought that each book has it's own id. Why is it nullable?

File: CRM.Library.Services/RegisterService/UserService.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+namespace CRMLibrary.Services.RegisterService
+{
+    public class UserService : IUserService
+    {
+        public UserService(IUserRepository userRepository)
+        {
+            _repository = userRepository;
+        }
+
+        public Task<User> Add(string name, string email, string password)
```
Comment (tstrizhyk) [inline CRM.Library.Services/RegisterService/UserService.cs]: What if user creation will be extended? I mean we add additional fields, such as user address, birthday, gender etc. In this case `public Task<User> Add(string name, string email, string password)` signature needs to be changed. I suggest to use model in this method as a parameter: `public Task<User> Add(User user)`

File: CRM.Library.Services/RegisterService/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            _repository.Add(newUser);
+            _repository.Complete();
+
+            return Task.FromResult(newUser);
+        }
+
+        public Task<User> Authenticate(string email, string password)
+        {
+            var foundUser = _repository
+                .Find(user => user.Email.ToLower() == email.ToLower())
```
Comment (tstrizhyk) [inline CRM.Library.Services/RegisterService/UserService.cs]:
```suggestion
                .Find(user => string.Equals(user.Email, email, StringComparison.OrdinalIgnoreCase))
```

File: CRM.Library.Services/RegisterService/UserService.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                .FirstOrDefault();
+
+            if (foundUser != null)
+            {
+                if (foundUser.Password == HashString(password))
```
Comment (tstrizhyk) [inline CRM.Library.Services/RegisterService/UserService.cs]: not a good solution to hash password in service. So it will be passed as plain text from UI, through RES API. Easy to hack it

File: CRMLibrary.RESTAPI/Controllers/ValuesController.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CRMLibrary.RESTAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ValuesController : ControllerBase
```
Comment (tstrizhyk) [inline CRMLibrary.RESTAPI/Controllers/ValuesController.cs]: Will it be extended?

File: CRM.Library.Services/HomeService/BookService.cs (hunk 1 above)
Comment (IvashchenkoIvanIgorevich) [reply CRM.Library.Services/HomeService/BookService.cs]: I thought that IGenericRepository consists of main methods for all repositories, other interfaces IBookRepository, IUserRepository which inherited from IGenericRepository, will be specific methods for own repository.
image

**Comment by IvashchenkoIvanIgorevich** [comment]
I'll make some edits to the .gitigrone file

File: CRM.Library.Services/HomeService/BookService.cs (hunk 2 above)
Comment (IvashchenkoIvanIgorevich) [reply CRM.Library.Services/HomeService/BookService.cs]: I got IGenericRepository<T> where this method was and didn't delete nullable type. I'll do this on the next pull request, thank you.

File: CRM.Library.Services/RegisterService/UserService.cs (hunk 3 above)
Comment (IvashchenkoIvanIgorevich) [reply CRM.Library.Services/RegisterService/UserService.cs]: > What if user creation will be extended? I mean we add additional fields, such as user address, birthday, gender etc. In this case `public Task<User> Add(string name, string email, string password)` signature needs to be changed. I suggest to use model in this method as a parameter: `public Task<User> Add(User user)`

Greate, I understand it and I'll do this on the next pull request, thank you.
